src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/repository/DocumentRepository.kt

```diff
@@ -1,8 +1,17 @@
 package uk.gov.justice.digital.hmpps.deliusapi.repository
 
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
 import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.deliusapi.entity.Document
 
 @Repository
-interface DocumentRepository : JpaRepository<Document, Long>
+interface DocumentRepository : JpaRepository<Document, Long> {
+
+  @Query(
+    """SELECT d FROM Document d WHERE d.offenderId=:offenderId 
+    AND d.externalReference like :uuid 
```

Comment:
Should we update this to make use of the index?
```suggestion
    AND lower(substring(d.externalReference, -36, 36)) = lower(:uuid)
```